Use thread::sleep instead of deprecated sleep_ms

jsanders opened the pull request:
Similarly, change one instance of `thread::park_timeout_ms`.

Fixes #8694

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/servo/servo/8705)

<!-- Reviewable:end -->

jsanders committed:
Use thread::sleep instead of deprecated sleep_ms

Similarly, change one instance of `thread::park_timeout_ms`.

Fixes #8694

metajack commented:
A small refactoring is suggesting. Otherwise r=me.

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4BHnqZPSRdl8D5RRz3:570197562) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
Can we get rid of this duplicated code by refactoring this logic into a helper function?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705#-:-K4BHxmqTuVPnUHHCfv3:2070515030)*
<!-- Sent from Reviewable.io -->

jsanders commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4DwVA4ABbhMVTmewWy:1263416748) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
I'd love to do that, but I wasn't sure where something like that would go. I was considering `util::task` or making a new `util` module in `profile`.  Any suggestions?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*
<!-- Sent from Reviewable.io -->

Wafflespeanut commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4EhlqkhkyDk7_XQ5Wi:-106037974) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
`util::task` is currently being used for spawning threads, IIRC. I guess it'd be better if we just create a new module in `util` (say, `time.rs`), and put our function there.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*
<!-- Sent from Reviewable.io -->

jsanders commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4EjiXbA1H4ZGP2iFOC:1396299650) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
I also looked into parsing the profiler options directly into `Some<Duration>` instead of `Some<f64>`, but that won't work unless I add a `serde::Serializer` implementation, which I'm not sure is the right approach.

Adding `util::time` seems reasonable.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*
<!-- Sent from Reviewable.io -->

jsanders commented:
Review status: 6 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4ErZnH541kzaRs0tLa:257465616) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
I went ahead and pulled this logic out as `util::time::duration_from_seconds`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*
<!-- Sent from Reviewable.io -->

Wafflespeanut commented:
@jsanders sgtm, squash the commits, btw?

Wafflespeanut commented on components/compositing/scrolling.rs:
```diff
@@ -55,9 +57,13 @@ impl ScrollingTimerProxy {
+            let delta_secs = delta_ns / NANOS_PER_SEC;
+            let delta_nanos = delta_ns % NANOS_PER_SEC;
+            assert!(delta_nanos <= u32::MAX as u64);
+            thread::sleep(Duration::new(delta_secs, delta_nanos as u32));
```
I think we could also have a function for this in `util/time` (for future usage), which takes nanoseconds as input and returns a duration.

jsanders replied:
Good idea. Note that this code is also very similar to what `Duration::new` already does to normalize the given nanos to be sub-second, but it only takes a `u32`, which `delta_ns` might not fit into.

Wafflespeanut commented on components/util/time.rs:
```diff
@@ -0,0 +1,20 @@
+use std::{u32, u64};
+
+pub fn duration_from_seconds(secs: f64) -> Duration {
+    const NANOS_PER_SEC: f64 = 1_000_000_000.0;
```
We could just pull the constant out, so that it'll be useful for future methods.

jsanders replied:
I wondered about that. It exists in `Duration` itself (well, as a `u32` rather than an `f64`), but it isn't public.

Wafflespeanut commented:
Once those changes have been made, squash the commits, and let's happily land this :)

jsanders commented:
I'm not familiar with your squashing policy. Should I squash all of the commits, or just the ones that I messed up by leaving the file out?

KiChjang commented:
@jsanders See https://github.com/servo/servo/wiki/Github-&-Critic-PR-handling-101. Ignore everything that has to do with critic; it's an old review system that we don't use anymore.

jsanders commented:
@KiChjang: Thanks.

jsanders commented:
@Wafflespeanut: Moved the nanoseconds-to-duration logic to `util::time`, fixed a couple off-by-one errors in the assertions in `duration_from_seconds`, and squashed.

Wafflespeanut commented:
Awesome! Thanks!

@Manishearth r+ this?

---

Reviewed 2 of 4 files at r4.
Review status: 6 of 10 files reviewed at latest revision, 2 unresolved discussions.

---

*[components/util/time.rs, line 9 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4Ezvv1I1RVB0Us2h-Q-r3-9:-K4LD4gzXXzYybXVPxnj:-1434150410) ([raw file](https://github.com/servo/servo/blob/a8e6aef1b1114895099b526f1c2744451d461c4f/components/util/time.rs#L9)):*
That's okay. This will be useful :)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705#-:-K4LDFf3zaULSVDuPxFd:796626617)*
<!-- Sent from Reviewable.io -->

Manishearth commented:
@bors-servo r=metajack,Wafflespeanut

bors-servo commented:
:pushpin: Commit 3659218 has been approved by `metajack,Wafflespeanut`

<!-- @bors-servo r=metajack,Wafflespeanut 3659218c59ef37d6e1aea202e0d6fa64275ede89 -->

bors-servo commented:
:hourglass: Testing commit 3659218 with merge 0f72049...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/697), [gonk](http://build.servo.org/builders/gonk/builds/712), [linux-dev](http://build.servo.org/builders/linux-dev/builds/694), [linux-rel](http://build.servo.org/builders/linux-rel/builds/774), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/717), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/704), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/729)
